sammba/modality_processors/perfusion_fair.py

```diff
@@ -0,0 +1,387 @@
+    # https://stackoverflow.com/a/45276885
+    
+    # cannot use a with statement in python 2.7; may cause problems
+    pool = Pool(processes=ncpu)
+    return list(tqdm.tqdm(pool.imap(partial(perf_fair_fitter,
+                                            t1_blood=t1_blood,
+                                            ti=ti,
+                                            t1_guess=t1_guess,
+                                            picker_sel=picker_sel,
+                                            picker_nonsel=picker_nonsel,
+                                            **kwargs),
+                                    all_s0), total = len(all_s0)))
+
+
+def perf_fair_nii_proc(nii_in_fname, t1_blood, ti, t1_guess, picker_sel,
+                       picker_nonsel, nii_out_fname=None, **kwargs):
+    """
+    Wrapper to execute perf_fair_fitter_mp on a NIfTI-1 image.
+
+    Parameters
+    ----------
+    nii_in_fname : str
+        Input file path.
+
+    t1_blood : int or float
+        T1 of blood in ms at the acquisition field strength. Empirical
+        constant; an example value is 2800 at 11.7T
+        (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3752414/table/T1,
+        Table 1 of Blood longitudinal (T1) and transverse (T2) relaxation time
+        constants at 11.7 Tesla, MAGMA. 2012 Jun; 25(3): 245–249,
+        Ai-Ling Lin, Qin Qin, Xia Zhao, and Timothy Q. Duong).
+
+    ti : list of int or float
+        The inversion times. Must have the same length as s0.
+
+    t1_guess : int or float
+        An initial starting value for T1. The mean of TIs is a good guess.
+
+    picker_sel : list or numpy array of int
+        Vector indicating positions of selectively inverted signals in s0. The
+        lengths of picker_sel and picker_sel must sum to the length of s0.
+
+    picker_nonsel : list or numpy array of int
+        Vector indicating positions of non-selectively inverted signals in s0.
+        The lengths of picker_sel and picker_sel must sum to the length of s0.
+
+    nii_out_fname : str
+        Output file path. If none, will be the same as the input file path,
+        but suffixed with _proc.
+
+    kwargs : dict, optional
+        Additional keyword arguments passed to perf_fair_fitter_mp.
+
+    Returns
+    -------
+    NIfTI-1 file saved to nii_out_fname.
+    """
+
+    nii_in = nib.load(nii_in_fname)
+    in_mat = nii_in.get_data()
+    all_s0 = in_mat.reshape((np.product(in_mat.shape[:-1]), in_mat.shape[-1]))
+    r = perf_fair_fitter_mp(all_s0, t1_blood, ti, t1_guess, picker_sel,
+                            picker_nonsel, **kwargs)
+    r = np.array(r)
+    img = nib.Nifti1Image(np.reshape(r, in_mat.shape[:-1] + (r.shape[1],)),
+                          nii_in.get_affine())
+    
+    if nii_out_fname is None:
+        nii_out_fname = nii_in_fname.replace('.nii.gz', '_proc.nii.gz')
+    return img.to_filename(nii_out_fname)
+
+
+def perf_fair_niiptbl_proc(nii_in_fname, t1_blood, **kwargs):

```

Comment:
if this is the function you will expose to the user, I think you need to explain what is the output in its documentation. The extensive explanations (CBF etc) should go rather in this exposed function than in the others